source/src/BNFC/Backend/Haskell/CFtoHappy.hs

```diff
@@ -66,7 +67,14 @@ header modName absName lexName tokenText eps = unlines $ concat
     , "{-# LANGUAGE PatternSynonyms #-}"
     , ""
     , "module " ++ modName
-    , "  ( happyError"
+    , "  ( Err"
+    , "  , Failure(..)"
```

Comment:
Maybe avoid exporting `Failure` & co if `--errors=string`? It might break the backward compatibility if the user imports `Par.hs` unqualified without import list.